pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -660,51 +655,71 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
 //    created.
 // 2) The certificate secret needed for Elasticsearch (in the Elasticsearch namespace).
 // 3) The certificate secret created by the ECK operator using the secret in 2).
-func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context) (*corev1.Secret, *corev1.Secret, *corev1.Secret, error) {
+func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (oprKeyCert *corev1.Secret, esKeyCert *corev1.Secret, oprCert *corev1.Secret, err error) {
 	svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
 
 	// Get the secret - might be nil
-	certSecret, err := utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, rmeta.OperatorNamespace())
+	oprKeyCert, err = utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, rmeta.OperatorNamespace())
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
 	// Ensure that cert is valid.
-	certSecret, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchCertSecret, certSecret, "tls.key", "tls.crt", rmeta.DefaultCertificateDuration, svcDNSNames...)
+	oprKeyCert, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchCertSecret, oprKeyCert, corev1.TLSPrivateKeyKey, corev1.TLSCertKey, rmeta.DefaultCertificateDuration, svcDNSNames...)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	certSecretESCopy := rsecret.CopyToNamespace(render.ElasticsearchNamespace, certSecret)[0]
-
-	// Get the pub secret - might be nil
-	pubSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, render.ElasticsearchNamespace)
+	oprKeyCertIssuer, err := utils.GetCertificateIssuer(oprKeyCert.Data[corev1.TLSCertKey])
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
-	// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
-	// public secret so it can get recreated.
-	if pubSecret != nil {
-		operatorManaged, err := utils.IsOperatorManaged(certSecret, "tls.crt")
+	// If Certificate management is enabled, we only want to trust the CA cert and let private key generation being handled from an init container.
+	if instl.CertificateManagement != nil {
+		cmCa := instl.CertificateManagement.CACert
+		cmIssuer, err := utils.GetCertificateIssuer(cmCa)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		if cmIssuer != oprKeyCertIssuer && !utils.IsOperatorIssued(oprKeyCertIssuer) {
+			return nil, nil, nil, fmt.Errorf("certificate management does not support custom Elastic secrets, please delete secret %s/%s or disable certificate management", oprKeyCert.Namespace, oprKeyCert.Name)
```

Comment:
This is something the user will likely see so could you please say Elasticsearch and not Elastic.

Also, I'm not sure "custom Elastic secrets" is right, you mean "customer Elasticsearch certificates" right?

Reply:
I'm also wondering if it's a fair assumption to just overwrite the certificate or delete if they own it if they have enabled certification management. Enabling certificate management sort of singles user intent that they expect this certificate to be managed by the system they've configured. I'd run this by @tmjd though first.

Reply:
I mean the "custom" secrets.

> I'm also wondering if it's a fair assumption to just overwrite the certificate or delete if they own it if they have enabled certification management.
I thought about this quite a bit. I asked CS, Tesla and Solutions and it seems like nobody ever used the BYO secrets feature, which is a good thing. For the hypothetical scenario in which a customer did bring his own certificate, I am hesitant to delete it, as we cannot judge how hard it was to obtain that certificate for them. They may be disappointed to see being deleted. Logging an error seemed like the appropriate thing to do.

If @Brian-McM or @tmjd or anyone else reviewing this feel that deleting it is better, I can be pusuaded.

Reply:
I think erroring is the correct thing to do if the custom certs were provided and cert management is enabled.

Reply:
Naw I'm good with the error, it seems like the right thing to do.